### PR TITLE
#355 EventOutputFormat

### PR DESCRIPTION
**Pull request by vincentrussell**

### CODE REVIEW

File: store/event-store/src/main/java/org/calrissian/accumulorecipes/eventstore/impl/AccumuloEventStore.java [hunk 1]
```diff
@@ -301,4 +301,45 @@ public void flush() throws Exception {
     public CloseableIterable<String> getTypes(String prefix, Auths auths) {
         return helper.getKeyValueIndex().getTypes(prefix, auths);
     }
+
+    public static class Builder {
```
Comment (eawagner) [inline store/event-store/src/main/java/org/calrissian/accumulorecipes/eventstore/impl/AccumuloEventStore.java]: I actually like the builder idea, but it should probably be brought in as part of another MR.  It should also include the identical EntityStore.

Comment (vincentrussell) [inline store/event-store/src/main/java/org/calrissian/accumulorecipes/eventstore/impl/AccumuloEventStore.java]: Really?  You're that formal with the merge requests?   I didn't understand how related the EntityStore and the EventStore are.  I'll take another look.  I also fixed the compilation issues, but I'm having issues in my test at query time.

Comment (eawagner) [inline store/event-store/src/main/java/org/calrissian/accumulorecipes/eventstore/impl/AccumuloEventStore.java]: Not really that formal, just things get messy quickly if there are a bunch of one-offs. Considering how close the event and entity stores are API wise it makes sense to keep those in sync, but not that big a deal to include builders for all the stores.

File: commons/src/main/java/org/calrissian/accumulorecipes/commons/iterators/support/EventFields.java [hunk 2]
```diff
@@ -72,7 +72,13 @@ public EventFields read(Kryo kryo, Input input, Class<EventFields> eventFieldsCl
         int entries = intSerializer.read(kryo, input, Integer.class);
         for (int i = 0; i < entries; i++) {
             // Read in the key
-            String key = stringSerializer.read(kryo, input, String.class);
+            String key = null;
+            try {
+                key = stringSerializer.read(kryo, input, String.class);
+            } catch (Throwable e) {
+                System.out.println(input);
```
Comment (eawagner) [inline commons/src/main/java/org/calrissian/accumulorecipes/commons/iterators/support/EventFields.java]: No reason to use System.out.println()

Comment (vincentrussell) [inline commons/src/main/java/org/calrissian/accumulorecipes/commons/iterators/support/EventFields.java]: The system.out is what helped me find the EventFields.clear() bug.  I pushed it to make sure the tests still ran.  For some reason the tests failures are different for me on my mac than on travis.  This branch needs to be rebased anyway since it depends on the other issues that you just merged.

**Comment by vincentrussell** [comment]
this should be able to be merged in after #360 .  I removed all of the unnecessary stuff.

**Comment by eawagner** [comment]
Merged in #360, might need to rebase/merge with master so that the builder class is available for the tests to pass.

**Comment by vincentrussell** [comment]
ok.  Thanks.  I rebased it and It looks ok I think.

File: store/event-store/src/main/java/org/calrissian/accumulorecipes/eventstore/hadoop/EventOutputFormat.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        OutputConfigurator.setConnectorInfo(CLASS, job.getConfiguration(), principal, token);
+    }
+
+    public static void setTypeRegistry(Job job, Class<TypeEncoder<?, String>>... encoderClass) throws AccumuloSecurityException {
+        setTypeRegistry(job, Lists.newArrayList(encoderClass));
+    }
+
+    public static void setTypeRegistry(Job job, Collection<Class<TypeEncoder<?, String>>> encoderClasses) throws AccumuloSecurityException {
+        ArgumentChecker.notNull(encoderClasses);
+        String value = Joiner.on(",").join(Iterables.transform(encoderClasses, new Function<Class, String>() {
```
Comment (vincentrussell) [inline store/event-store/src/main/java/org/calrissian/accumulorecipes/eventstore/hadoop/EventOutputFormat.java]: Yea...that's clearly better.

**Comment by eawagner** [comment]
Would probably fix the copyright headers on the file to be 2016.

Otherwise +1, we should merge this in.
